src/integrationTest/java/uk/gov/hmcts/reform/cpo/repository/CasePaymentOrdersAuditTest.java

```diff
@@ -0,0 +1,151 @@
+
+import com.microsoft.applicationinsights.core.dependencies.apachecommons.lang3.RandomStringUtils;
+import org.hibernate.envers.AuditReader;
+import org.hibernate.envers.AuditReaderFactory;
+import org.hibernate.envers.RevisionType;
+import org.hibernate.envers.query.AuditEntity;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import uk.gov.hmcts.reform.cpo.BaseTest;
+import uk.gov.hmcts.reform.cpo.data.CasePaymentOrderEntity;
+import uk.gov.hmcts.reform.cpo.domain.CasePaymentOrder;
+import uk.gov.hmcts.reform.cpo.service.mapper.CasePaymentOrderMapper;
+
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.NoResultException;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@SuppressWarnings({"PMD.JUnitAssertionsShouldIncludeMessage", "PMD.DataflowAnomalyAnalysis"})
+public class CasePaymentOrdersAuditTest extends BaseTest {
+
+    @Autowired
+    private CasePaymentOrdersRepository casePaymentOrdersRepository;
+
+    @Autowired
+    private CasePaymentOrderMapper casePaymentOrderMapper;
+
+    @Autowired
+    EntityManagerFactory entityManagerFactory;
+
+    private CasePaymentOrderEntity casePaymentOrderEntity;
+
+    private AuditReader auditReader;
+
+    private static final String ACTION_VALUE = "Action";
+    private static final String NEW_ACTION_VALUE = "New Action";
+
+    @BeforeEach
+    void setUp() {
+        CasePaymentOrderEntity casePaymentOrderEntityToBePersisted =
+                casePaymentOrderMapper.toEntity(CasePaymentOrder.builder()
+                .orderReference(RandomStringUtils.random(10, true, true))
+                .action(ACTION_VALUE)
+                .caseId(new Random().nextLong())
+                .createdBy("gsullivan")
+                .effectiveFrom(LocalDateTime.now())
+                .responsibleParty("mrresponsible")
+                .build()
+        );
+        casePaymentOrderEntity = casePaymentOrdersRepository.save(casePaymentOrderEntityToBePersisted);
+        assertNotNull(casePaymentOrderEntity);
+
+        auditReader = AuditReaderFactory.get(entityManagerFactory.createEntityManager());
+    }
+
+    @Test
+    void testAuditOfCreatingCasePaymentOrder() {
+        CasePaymentOrderEntity auditedCasePaymentOrderEntity = (CasePaymentOrderEntity) auditReader
+                .createQuery()
+                .forRevisionsOfEntity(CasePaymentOrderEntity.class, true, true)
+                .add(AuditEntity.id().eq(casePaymentOrderEntity.getId()))
+                .getSingleResult();
```

Comment:
this API is a bit complicated to understand. To make unit tests more understandable, consider "hiding" this complexity behind a private method with a nice explanatory name

Reply:
It is a horrible API for sure - will do

Reply:
refactored in commit https://github.com/hmcts/cpo-case-payment-orders-api/pull/29/commits/2feddd456da4c714ca6caed9d8207ee8bc40cf46